EMS/core-bundle/src/DataTable/Type/UploadedFileLogDataTableType.php

```diff
@@ -0,0 +1,72 @@
+
+class UploadedFileLogDataTableType extends AbstractEntityTableType
+{
+    public function __construct(
+        FileService $entityService,
+        private readonly Security $security
+    ) {
+        parent::__construct($entityService);
+    }
+
+    public function build(EntityTable $table): void
+    {
+        $table->addColumnDefinition(new BoolTableColumn('uploaded-file.index.column.available', 'available'));
+        $table->addColumn('uploaded-file.index.column.name', 'name')
+            ->setRoute('ems_file_download', function (UploadedAsset $data) {
+                if (!$data->getAvailable()) {
+                    return null;
+                }
+
+                return [
+                    'sha1' => $data->getSha1(),
+                    'type' => $data->getType(),
+                    'name' => $data->getName(),
+                ];
+            });
+        $table->addColumnDefinition(new DatetimeTableColumn('uploaded-file.index.column.created', 'created'));
+        $table->addColumnDefinition(new UserTableColumn('uploaded-file.index.column.username', 'user'));
+        $table->addColumnDefinition(new BytesTableColumn('uploaded-file.index.column.size', 'size'));
+        $table->addColumnDefinition(new BoolTableColumn('uploaded-file.index.column.hidden', 'hidden'));
+        $table->addColumnDefinition(new DatetimeTableColumn('uploaded-file.index.column.head_last', 'headLast'));
+        $table->addColumn('uploaded-file.index.column.type', 'type');
+        $table->addColumn('uploaded-file.index.column.sha1', 'sha1');
+
+        $table->addTableAction(TableAbstract::DOWNLOAD_ACTION, 'fa fa-download', 'uploaded-file.uploaded-file.download_selected', 'uploaded-file.uploaded-file.download_selected_confirm');
+
+        if ($this->security->isGranted(Roles::ROLE_PUBLISHER)) {
+            $table->addDynamicItemPostAction('ems_core_uploaded_file_show_hide', 'uploaded-file.action.hide-show', 'eye', 'uploaded-file.hide-show-confirm', ['assetId' => 'id']);
+            $table->addTableAction(UploadedFileController::HIDE_ACTION, 'fa fa-eye', 'uploaded-file.uploaded-file.hide-show', 'uploaded-file.uploaded-file.hide-show_confirm');
+        }
+        if ($this->security->isGranted(Roles::ROLE_ADMIN)) {
+            $itemDeleteAction = $table->addDynamicItemPostAction('ems_file_soft_delete', 'uploaded-file.action.soft-delete', 'trash', 'uploaded-file.soft-delete-confirm', ['id' => 'id']);
+            $itemDeleteAction->setButtonType('outline-danger');
+            $deleteAction = $table->addTableAction(UploadedFileController::SOFT_DELETE_ACTION, 'fa fa-trash', 'uploaded-file.uploaded-file.soft_delete_selected', 'uploaded-file.uploaded-file.soft_delete_selected_confirm');
+            $deleteAction->setCssClass('btn btn-outline-danger');
+        }
+
+        $table->setDefaultOrder('created', 'desc');
+    }
+
+    public function getRoles(): array
+    {
+        return [Roles::ROLE_ADMIN];
```

Comment:
I'm seeing ROLE_Publisher in the build function. I guess that this table is at least for publuisher

Reply:
It's not visible for publishers, the route is prefix wit `/admin` and it's in in the admin menu section

Reply:
Fixed in [bbb0673](https://github.com/ems-project/elasticms/pull/493/commits/bbb0673925745d85650838eb2de3614609ef98c8)